protocols/translation/decoder.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation
+
+import "go.opentelemetry.io/collector/consumer/pdata"
+
+type MetricsDecoder interface {
+	// ToMetrics converts a data model of another protocol into pdata.
+	ToMetrics(src interface{}) (pdata.Metrics, error)
+	// Type returns an instance of the model.
```

Comment:
Hmm, this is a bit unexpected. Why is this named `Type`? Should it be `New` or something?

Reply:
Hmm. Yeah I called it Type because you want to know it's type but don't really care about the instance itself. Maybe `NewModel` would be more clear?

Reply:
What's the intended use? Maybe drop it from this PR and add later when the usage is clear.

Reply:
Removed as part of returning interface{}